docs/building.md

```diff
@@ -222,27 +222,66 @@ cargo run --release --bin slint-viewer -- examples/printerdemo/ui/printerdemo.sl
 
 ## Generating the Documentation
 
-### Rust
+The Slint documentation consists of four parts:
 
-The documentation for the different crates is built using rustdoc.
+- The Slint tutorials
+- The Slint Rust DSL documentation
+- The Slint C++ API documentation
+- The Slint Node.js API documentation
 
-The language documentation has snippets in the .slint language which can be previewed by injecting
-html to the documentation with the `--html-in-header` rustdoc flag.
+### Prerequisites
 
-Use the following command line to build the documentation to include preview of the .slint files.
+- [mdbook](https://rust-lang.github.io/mdBook/guide/installation.html)
+- [pipenv](https://pipenv.pypa.io/en/latest/)
+- [Python](https://www.python.org/downloads/)
+- [Doxygen](https://www.doxygen.nl/download.html)
 
-```sh
-RUSTDOCFLAGS="--html-in-header=$PWD/docs/resources/slint-docs-preview.html --html-in-header=$PWD/docs/resources/slint-docs-highlight.html" cargo doc --no-deps --features slint/document-features,slint/log
```

Comment:
This command is "lost", but it's needed to build the Rust API docs.

Reply:
Ahh, this is part of the confusion between DSL and API, will fix.

Reply:
Once upon a time, there was no Slint DSL documentation. Instead, everything was in the C++ or Rust docs, and the Slint DSL docs was part of it.
For example https://archive.slint.dev/releases/0.3.5/docs/rust/slint/docs/langref/   and https://archive.slint.dev/releases/0.3.5/docs/cpp/language
(The preview no longer works on the old docs because of domain change: `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://slint.dev/releases/0.3.5/wasm-interpreter/slint_wasm_interpreter.js. (Reason: CORS request did not succeed).`  But that's old docs anyway.


It was then decided to merge them because we were not fan of the duplication.

But now, the slint-docs-preview.html doesn't really make sense anymore since there are no slint code sample in the rust doc.

Anyway, this was just some piece of history